sdks/python/src/opik/api_objects/dataset/utils.py

```diff
@@ -0,0 +1,24 @@
+import json
+import hashlib
+from typing import Dict, Any, Union
+from . import dataset_item
+
+
+def compute_content_hash(item: Union[dataset_item.DatasetItem, Dict[str, Any]]) -> str:
+    if isinstance(item, dataset_item.DatasetItem):
+        content = {
+            "input": item.input,
+            "expected_output": item.expected_output,
+            "metadata": item.metadata,
+        }
+    else:
+        content = item
+
+    # Convert the dictionary to a JSON string with sorted keys for consistency
+    json_string = json.dumps(content, sort_keys=True)
```

Comment:
I’d like to point out that if the value for the key is a collection, such as an array or a set, and the order of elements is different, the items will be considered different even though their content is nearly identical. Though it might be not relevant to our use-cases.